<launch>
  <!-- Input -->
  <arg name="input_autoware_engage" default="/vehicle/engage" />
  <arg name="input_vehicle_state_report" default="/vehicle/state_report" />
  <arg name="input_is_emergency" default="/vehicle/emergency/is_emergency" />
  <arg name="input_route" default="/planning/global_path" />
  <arg name="input_odometry" default="/vehicle/odometry" />

  <!-- Output -->
  <arg name="output_autoware_state" default="/autoware/state" />
  <arg name="output_autoware_engage" default="/vehicle/engage" />

  <!-- Service -->
  <arg name="service_shutdown" default="/autoware/shutdown" />
  <arg name="service_reset_route" default="/autoware/reset_route" />

  <!-- Config -->
  <arg name="config_file" default="$(find-pkg-share autoware_state_monitor)/config/autoware_state_monitor.param.yaml" />

  <!-- Parameter -->
  <arg name="update_rate" default="10.0" />
  <arg name="th_arrived_distance_m" default="2.0" />
  <arg name="th_stopped_time_sec" default="1.0" />
  <arg name="th_stopped_velocity_mps" default="0.01" />
  <arg name="disengage_on_route" default="true" />
  <arg name="disengage_on_goal" default="true" />

  <node pkg="autoware_state_monitor" exec="autoware_state_monitor" name="autoware_state_monitor" output="screen">
    <param name="use_sim_time" value="$(env AW_ROS2_USE_SIM_TIME false)"/>
    <remap from="input/autoware_engage" to="$(var input_autoware_engage)"/>
    <remap from="input/vehicle_state_report" to="$(var input_vehicle_state_report)"/>
    <remap from="input/is_emergency" to="$(var input_is_emergency)"/>
    <remap from="input/route" to="$(var input_route)"/>
    <remap from="input/odometry" to="$(var input_odometry)"/>

    <remap from="output/autoware_state" to="$(var output_autoware_state)"/>
    <remap from="output/autoware_engage" to="$(var output_autoware_engage)"/>

    <remap from="service/shutdown" to="$(var service_shutdown)"/>
    <remap from="service/reset_route" to="$(var service_reset_route)"/>

    <param from="$(var config_file)" />

    <param name="update_rate" value="$(var update_rate)" />
    <param name="th_arrived_distance_m" value="$(var th_arrived_distance_m)" />
    <param name="th_stopped_time_sec" value="$(var th_stopped_time_sec)" />
    <param name="th_stopped_velocity_mps" value="$(var th_stopped_velocity_mps)" />
    <param name="disengage_on_route" value="$(var disengage_on_route)" />
    <param name="disengage_on_goal" value="$(var disengage_on_goal)" />
  </node>
</launch>
